src/argilla/client/nightly.py

```diff
@@ -0,0 +1,231 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+import warnings
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, TypeVar, Union
+
+from pydantic import BaseModel, create_model
+from tqdm import tqdm
+
+import argilla as rg
+
+if TYPE_CHECKING:
+    from argilla.client.api import Argilla
+    from argilla.client.sdk.datasets.v1.models import FeedbackDataset
+
+FieldsSchema = TypeVar("FieldsSchema", bound=BaseModel)
+
+
+class RecordSchema(BaseModel):
+    fields: FieldsSchema
+    response: Optional[Dict[str, Any]] = {"values": {}}
+    external_id: Optional[str] = None
+
+
+class OnlineRecordSchema(RecordSchema):
+    id: str
+    inserted_at: Optional[str] = None
+    updated_at: Optional[str] = None
+
+
+class DatasetStatus(str, Enum):
+    DRAFT = "draft"
+    READY = "ready"
+
+
+class QuestionSchema(BaseModel):
+    name: str
+    title: str
+    required: bool
+    settings: Dict[str, Any]
+
+
+class Dataset:
```

Comment:
I'm planning to combine the current with some ideas that @frascuchon left at https://gist.github.com/frascuchon/2e11db8a00559746e4b4c2dff687daaf